src/components/alarm/AlarmController.cpp

```diff
@@ -88,21 +103,66 @@ uint32_t AlarmController::SecondsToAlarm() {
 
 void AlarmController::DisableAlarm() {
   xTimerStop(alarmTimer, 0);
-  state = AlarmState::Not_Set;
+  isAlerting = false;
+  if (alarm.isEnabled) {
+    alarm.isEnabled = false;
+    SaveAlarmToFile();
+  }
 }
 
 void AlarmController::SetOffAlarmNow() {
-  state = AlarmState::Alerting;
+  isAlerting = true;
   systemTask->PushMessage(System::Messages::SetOffAlarm);
 }
 
 void AlarmController::StopAlerting() {
-  // Alarm state is off unless this is a recurring alarm
-  if (recurrence == RecurType::None) {
-    state = AlarmState::Not_Set;
+  isAlerting = false;
+  // Disable alarm unless it is recurring
+  if (alarm.recurrence == RecurType::None) {
+    alarm.isEnabled = false;
+    SaveAlarmToFile();
   } else {
     // set next instance
     ScheduleAlarm();
   }
   systemTask->PushMessage(System::Messages::StopRinging);
 }
+
+void AlarmController::SetRecurrence(RecurType recurrence) {
+  if (alarm.recurrence != recurrence) {
+    alarm.recurrence = recurrence;
+    SaveAlarmToFile();
+  }
+}
+
+void AlarmController::LoadAlarmFromFile() {
+  lfs_file_t file;
+  AlarmData buffer;
+
+  if (fs.FileOpen(&file, "/alarm.dat", LFS_O_RDONLY) != LFS_ERR_OK) {
+    NRF_LOG_WARNING("[AlarmController] Failed to open alarm data file");
+    return;
+  }
+
+  fs.FileRead(&file, reinterpret_cast<uint8_t*>(&buffer), sizeof buffer);
+  fs.FileClose(&file);
+  if (buffer.version != alarmFormatVersion) {
+    NRF_LOG_WARNING("[AlarmController] Loaded alarm data has version %u instead of %u, discarding", buffer.version, alarmFormatVersion);
+    return;
+  }
+
+  alarm = buffer;
+  NRF_LOG_INFO("[AlarmController] Loaded alarm data from file");
+}
+
+void AlarmController::SaveAlarmToFile() {
+  lfs_file_t file;
```

Comment:
nitpick: rename to `alarmFile` to be more explicit which file is meant (to be closer to the `Settings::SaveSettingsFromFile()` function)

```suggestion
  lfs_file_t alarmFile;
```

Reply:
I got derailed trying to pull the save/load functions out to avoid duplicating them and that was a left-over :) IMHO worth doing at some point, but one thing at a time.